Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -299,11 +299,30 @@ function Wait-ServiceFabricApplicationUpgradeAction
     }
 
     $upgradeStatusValidator = { param($upgradeStatus) return !($upgradeStatus.UpgradeState -eq "RollingBackCompleted" -or $upgradeStatus.UpgradeState -eq "RollingForwardCompleted") }
+
+    $exceptionRetryEvaluator = {
+        param($ex)
+
+        if ($ex.GetType().FullName -eq "System.Fabric.FabricException")
+        {
+            if($ex.ErrorCode -eq "InvalidCredentials")
+            {
+                $clusterConnectionParameters = @{}
+                $connectedServiceEndpoint = Get-VstsEndpoint -Name $serviceConnectionName -Require
+                Connect-ServiceFabricClusterFromServiceEndpoint -ClusterConnectionParameters $clusterConnectionParameters -ConnectedServiceEndpoint $connectedServiceEndpoint
+                return $true
+            }
+            return $false
+        }
+        return $true
+    }
+
     $upgradeStatus = Invoke-ActionWithRetries -Action $upgradeStatusFetcher `
```

Comment:
Do we not need to handle this for other operations

Reply:
We don't see any potential operation which can take over 1 hour. Also its never been reported.
So, I don't think its justified to add the same for other operations like Copy/Register which might take 1 hour but for that case the package size need to be exceptionally high. I believe we can disregard such possibility and lets don't handle the same for other operation.

Reply:
Makes sense